field_RANKSUFF_module.fypp

```diff
@@ -237,26 +237,31 @@ CONTAINS
       IF (ZERO) VIEW_PTR(${ft.viewShape}$) = ${ft.default}$
     END IF
 
+    IF (ASSOCIATED (SELF%PARENT)) THEN
+      CALL SELF%PARENT%SET_DEVICE_DIRTY ()
+    ELSE
 ! Mark data dirty on the device: the pointer returned by GET_VIEW may be rw
-    IF (IAND (SELF%ISTATUS, NDEVFRESH)==NDEVFRESH) THEN
-      SELF%ISTATUS = IAND(SELF%ISTATUS, NOT(NDEVFRESH))
+      CALL SELF%SET_DEVICE_DIRTY ()
```

Comment:
I could be mistaken, but I think here we do not mark the children as dirty on the device if `GET_VIEW` is called on the gang. I think the children should also be marked as dirty.

Reply:
The code has moved, but I think SET_DEVICE_DIRTY will call SET_STATUS, which will also update children status.